libraries/shared/src/shared/Storage.cpp

```
@@ -90,3 +90,31 @@ FileStorage::~FileStorage() {
         _file.close();
     }
 }
+
+void FileStorage::ensureWriteAccess() {
+    if (_hasWriteAccess) {
+        return;
+    }
+
+    if (_mapped) {
+        if (!_file.unmap(_mapped)) {
+            throw std::runtime_error("Unable to unmap file");
+        }
+    }
+    if (_file.isOpen()) {
+        _file.close();
+    }
+    _valid = false;
+
+    if (_file.open(QFile::ReadWrite)) {
+        _mapped = _file.map(0, _file.size());
+        if (_mapped) {
+            _valid = true;
+            _hasWriteAccess = true;
+        } else {
+            qCWarning(storagelogging) << "Failed to map file " << _file.fileName();
+        }
+    } else {
+        qCWarning(storagelogging) << "Failed to open file " << _file.fileName();
```

Should we throw an exception in these cases? This will almost definitely cause issues later.